api/v1/models/labels.go

```diff
@@ -25,3 +27,22 @@ func (m Labels) Validate(formats strfmt.Registry) error {
 	}
 	return nil
 }
+
+// GetMap returns a map with the Labels in key-value format. If one label
+// cannot be in the key-value format will be skipped on the output.
+func (m Labels) GetMap() map[string]string {
```

Comment:
Since this is k8s specific, it's worth just copying over the k8s labels. The controller has RLock access to the endpoint, and it might be easier to iterate over the `labels` field directly and only copy over labels with source k8s.